maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/VersionsHelper.java

```diff
@@ -80,11 +81,15 @@ private void sortDescending(List<ArtifactVersion> versions){
      * @return inferred version for given artifact
      */
     public String inferVersion(Artifact artifact){
-        List<ArtifactVersion> artifactVersions = getVersions(artifact);
-        if(artifactVersions.size() == 0){
+        if (new Version(artifact.getVersion()).isSnapshot()) {
+            List<ArtifactVersion> artifactVersions = getVersions(artifact);
+            if(artifactVersions.size() == 0){
+                return artifact.getVersion();
+            }
+            return inferVersion(artifact.getVersion(), artifactVersions);
```

Comment:
@AdamGrzybkowski I don't have enough knowledge about all the callers of this method to answer your question. I said what is the expected behavior of this particular feature so go ahead and decide what changes are needed.